weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallChat.tsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * Code to determine if a call is a chat call, and if so,
+ * load the message data and render it.
+ */
+
+import _ from 'lodash';
+import React, {useEffect, useMemo, useState} from 'react';
+
+import {LoadingDots} from '../../../../../LoadingDots';
+import {isWeaveRef} from '../../filters/common';
+import {MessageList} from '../PromptPage/MessageList';
+import {useWFHooks} from '../wfReactInterface/context';
+import {
+  CallSchema,
+  Loadable,
+} from '../wfReactInterface/wfDataModelHooksInterface';
+
+type CallChatProps = {call: CallSchema};
+
+// Does this look like a message object?
+export const isMessage = (message: any): boolean => {
+  if (!_.isPlainObject(message)) {
+    return false;
+  }
+  // TODO: Check for role: string?
+  if (!('content' in message) && !('tool_calls' in message)) {
+    return false;
+  }
+  return true;
+};
+
+// Does this call look like a chat formatted object?
+export const isCallChat = (call: CallSchema): boolean => {
+  if (!('messages' in call.rawSpan.inputs)) {
+    return false;
+  }
+  const {messages} = call.rawSpan.inputs;
+  if (!_.isArray(messages)) {
+    return false;
+  }
+  return messages.every(isMessage);
+};
+
+const useRefData = (refUri: string): Loadable<any> => {
+  const {useRefsData} = useWFHooks();
+  const refsData = useRefsData(refUri ? [refUri] : []);
+  return useMemo(() => {
+    const {loading, result} = refsData;
+    const firstResult = result ? result[0] : null;
+    return {loading, result: firstResult};
+  }, [refsData]);
+};
+
+export const CallChat = ({call}: CallChatProps) => {
+  const {inputs, output} = call.rawSpan;
+
+  const {messages} = inputs;
+  const inputRefs = messages
+    .map((m: {content: any}) => (isWeaveRef(m.content) ? m.content : null))
+    .filter(Boolean);
+  const {useRefsData} = useWFHooks();
+  const inputData = useRefsData(inputRefs);
+
+  const outputData = useRefData(output?._result);
```

Comment:
This blindly assumes `output` is a ref. I think this and all other cases need to be adjusted. With Andrew's inlining stuff, it is not guaranteed to be a ref